### PR TITLE
Re-use fragments for shared element transition

### PR DESCRIPTION
**Pull request by fornewid**

### CODE REVIEW

**Comment by fornewid** [comment]
The FragmentStateChanger for shared elements is different with other FragmentStateChanger classes. (If re-enter GridFragment using goBack(), GridFragment are recreated.)
So match codes with others. (Use 'detach/attach', instead of 'hide/show'.)

**Comment by Zhuinden** [comment]
Hrmm I'll need to investigate that then, but using `add/attach/detach/remove` instead of `replace` didn't trigger shared element transition for me, which is why I used `replace` between the top and the fragment below it.

I did NOT try if setting `setReorderingAllowed(true)` would alter this behavior or not.

I'll check this later

File: simple-stack-example-sharedelement-fragment/src/main/java/com/example/fragmenttransitions/FragmentStateChanger.java [hunk 1]
```diff
@@ -66,27 +66,29 @@ public void handleStateChange(StateChange stateChange) {
...
+                if(!newState.contains(oldKey)) {
                     fragmentTransaction.remove(fragment);
+                } else if(!fragment.isDetached()) {
+                    fragmentTransaction.detach(fragment);
                 }
             }
         }
         for(BaseKey newKey : newState) {
             Fragment fragment = fragmentManager.findFragmentByTag(newKey.getFragmentTag());
-            if(!newKey.equals(topNewKey) && !newKey.equals(topPreviousKey)) {
```
Comment (Zhuinden) [inline simple-stack-example-sharedelement-fragment/src/main/java/com/example/fragmenttransitions/FragmentStateChanger.java]: Yeah I can see that being a bug potentially on my side

**Comment by fornewid** [comment]
Thanks for your response. :)
I'll also recheck transition on the other devices.

**Pull request merged**